src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Yevgeniy Zholkevskiy <zhenya.zholkevskiy@gmail.com>
+ */
+class UniqueValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Unique) {
+            throw new UnexpectedTypeException($constraint, Unique::class);
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        if (!$value instanceof \IteratorAggregate) {
```

Comment:
is_array part must stay, I wanted to change second part of condition only. This won't work with arrays now.